rust/error-tracking/src/team_cache.rs

```diff
@@ -0,0 +1,58 @@
+use std::{future::Future, time::Duration};
+
+use common_types::Team;
+use moka::sync::{Cache, CacheBuilder};
+use tokio::sync::RwLock;
+
+use crate::error::Error;
+
+// This /could/ be moved into common/types, but I'd have to take
+// a dependency on tokio and moka to do it, and I don't want to
+// do that just yet (although it is used across here and feature
+// flags, so....)
+pub struct TeamCache {
```

Comment:
Yeah that's exactly what I'd do - there's a way to make this Generic across a T, but it's up to you whether you want to do that or copy-paste.

Reply:
The only thing to keep in mind is that it's really nice to have the memory usage per-cached-item be stable, as otherwise you'll have a hard time predicting how much RAM your service will use